stdlib/public/core/Integers.swift.gyb

```diff
@@ -424,46 +1729,202 @@ public func > <T : BinaryInteger>(lhs: T, rhs: T) -> Bool {
 //===--- FixedWidthInteger ------------------------------------------------===//
 //===----------------------------------------------------------------------===//
 
+/// An indicator of whether an arithmetic operation overflowed.
+///
+/// Some arithmetic operations on fixed-width integers return an
+/// `ArithmeticOverflow` instance to indicate whether an overflow has
+/// occurred. For example, adding `UInt8.max` to itself results in a value that
+/// can't be represented by an `UInt8` instance without overflowing.
+///
+///     let x = UInt8.max
+///     // x == 255
+///     let (y, overflow) = x.addingWithOverflow(x)
+///     // y == 254
```

Comment:
I've documented the behavior on the methods that produce an overflow, e.g. https://github.com/natecook1000/swift/blob/a4bdcd5e5117b428c9948e6aa0f56bc28261a64e/stdlib/public/core/Integers.swift.gyb#L453-L458 — is that sufficient? if we need further documentation of the guarantee I would add it to the `FixedWidthInteger` discussion, not here.